stdlib/importlib/metadata/__init__.pyi

```diff
@@ -35,6 +35,23 @@ if sys.version_info >= (3, 8):
             dist: ClassVar[Distribution | None]
             def matches(self, **params: Any) -> bool: ...  # undocumented
 
+    if sys.version_info >= (3, 10):
+        class EntryPoints(list[EntryPoint]):  # use as list is deprecated since 3.10
+            def select(self, **params: Any) -> EntryPoints: ...
+            @property
+            def names(self) -> set[str]: ...
+            @property
+            def groups(self) -> set[str]: ...
+
+        class SelectableGroups(dict[str, EntryPoints]):  # use as list is deprecated since 3.10
+            @classmethod
+            def load(cls, eps: SelectableGroups) -> SelectableGroups: ...
+            @property
+            def groups(self) -> set[str]: ...
+            @property
+            def names(self) -> set[str]: ...
+            def select(self, **params: Any) -> SelectableGroups: ...
```

Comment:
I'm curious, shouldn't it be only `str` and not `Any`? I think it can only be `name`, `group`, `value`, `module`, `attr` and `extras` here, all are just strings as far as I understand?

Reply:
It looks like `extras` is a list of strings instead, so `str` doesn't seem right. However, we could just specify all of the allowed attributes manually to provide more precise types.

Reply:
Added.